ideascube/static/ideascube/js/ideascube.js

```diff
@@ -265,3 +265,26 @@ ID.confirmClick = function (selector) {
     };
     el.addEventListener('click', ask, false);
 };
+
+
+ID.initWifiList = function(item_selector, popup_selector, urlroot) {
+    var elements = document.querySelectorAll(item_selector);
+
+    for (var element of elements) {
+        var known = element.getAttribute('data-known') === 'True';
+        var secure = element.getAttribute('data-secure') === 'True';
+
+        if (known || !secure) {
+            var ssid = element.getAttribute('data-ssid');
+            element.setAttribute('href', urlroot + ssid);
+        } else {
+            element.setAttribute('href', popup_selector);
+
+            element.addEventListener('click', function(evt) {
```

Comment:
> In general, I'm a bit cautious about making reusable functions before there's actually a place where it would be nice to reuse it

Agreed :)

> Can't we wait for having a need to display other popups, and at that point we can change that code, make it reusable, and reuse it?

I already have those use case, but OK let's move as is, I'll refactor in a later step :)